### ISSUE TITLE
Diff of issue view wrong after rebase

### ISSUE THREAD

**Issue by alexanderlink** [reviewninja/review.ninja#853]
Assume a pull request from branch x (x_commit_a) to master (master_commit_a).
After some time I rebased x to the latest master head commit (master_commit_a+n) and did a force push to x in order to have a cleaner/shorter merge commit history when the pull request is merged.

In the issue view the displayed diff now becomes very large since it also contains all commits between master_commit_a and master_commit_a+n.

I assume the reason is the commits on branch x changed from x_commit_a to x_commit_a2 and the issue was created on x_commit_a. The diff between x_commit_a and x_commit_a2 now of course contains all commits from master_commit_a to master_commit_a+n as well.

Review Ninja should somehow recognize force pushes rewriting the history. Or it could recognize this situation and identify the corresponding new commits.

**Comment by sschuberth** [comment]
I fully agree that dealing with (interactively) rebased branches could be improved a lot. I started to look at ReviewNinja in the hope it would fix the issues with force-pushed branches that GitHub review has.

A typical scenario for us that we fixup existing commits to address issues instead of adding new commits. This keeps the history shorter and cleaner but requires us to force-push a new iteration of that branch.

With this workflow we would require ReviewNinja to be able to show the diff between the original branch and the new force-pushed version of the branch to check which issues have been addressed.
